extensions/transports-http-hc5/runtime/src/main/java/io/quarkiverse/cxf/transport/http/hc5/QuarkusWorkQueueImpl.java

```diff
@@ -0,0 +1,74 @@
+package io.quarkiverse.cxf.transport.http.hc5;
+
+import org.apache.cxf.common.classloader.ClassLoaderUtils;
+import org.apache.cxf.common.classloader.ClassLoaderUtils.ClassLoaderHolder;
+import org.apache.cxf.workqueue.AutomaticWorkQueue;
+import org.eclipse.microprofile.context.ManagedExecutor;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.InstanceHandle;
+
+/**
+ * Executes the tasks using {@link ManagedExecutor} for the sake of context propagation.
+ */
+public class QuarkusWorkQueueImpl implements AutomaticWorkQueue {
+
+    private final String name;
+
+    public QuarkusWorkQueueImpl(String name) {
+        this.name = name;
+    }
+
+    @Override
+    public void execute(final Runnable command) {
+        //Grab the context classloader of this thread.   We'll make sure we use that
+        //on the thread the runnable actually runs on.
+
+        final ClassLoader loader = Thread.currentThread().getContextClassLoader();
+        Runnable r = new Runnable() {
+            public void run() {
+                ClassLoaderHolder orig = ClassLoaderUtils.setThreadContextClassloader(loader);
+                try {
+                    command.run();
+                } finally {
+                    if (orig != null) {
+                        orig.reset();
+                    }
+                }
+            }
+        };
+        InstanceHandle<ManagedExecutor> managedExecutorInst = Arc.container().instance(ManagedExecutor.class);
+        if (managedExecutorInst.isAvailable()) {
+            ManagedExecutor managedExecutor = managedExecutorInst.get();
```

Comment:
@FroMage, one more question: I thought this would give me some global ManagedExecutor managed by Quarkus and configured using some standard Quarkus properties. But the more I read the [context propagations guide](https://quarkus.io/guides/context-propagation), the more it sounds like I would get a new instance every time I call `Arc.container().instance(ManagedExecutor.class)` - is that correct?

Reply:
In the context of Quarkus, you will get the Quarkus executor. At least that's the theory ;)

Reply:
Would that be the one configured through the quarkus.thread-pool.* set of options? https://quarkus.io/guides/all-config#quarkus-core_quarkus.thread-pool.core-threads

Reply:
Yes.

Reply:
Yes.

Reply:
Perfect, thanks a lot! I have adapted our docs: https://github.com/quarkiverse/quarkus-cxf/pull/957